src/AutoRest.CSharp/LowLevel/Decorator/MethodAmbiguityResolver.cs

```diff
@@ -0,0 +1,170 @@
+using AutoRest.CSharp.Output.Models.Types;
+using Microsoft.CodeAnalysis;
+
+namespace AutoRest.CSharp.LowLevel.Decorator
+{
+    internal static class MethodAmbiguityResolver
+    {
+        /// <summary>
+        /// Resolve ambiguous methods between DPG client and existing client codes.
+        /// </summary>
+        /// <param name="clientMethods">Client methods belonging to a DPG client.</param>
+        /// <param name="clientNamespace">Namespace of the DPG client.</param>
+        /// <param name="clientName">Name of the DPG client.</param>
+        /// <param name="sourceInputModel">Roslyn source model for existing client codes.</param>
+        /// <returns>An array of LowLwevelClientMethod which have been resolved to avoid ambiguity with existing client codes.</returns>
+        internal static LowLevelClientMethod[] ResolveClientMethods(LowLevelClientMethod[] clientMethods, string clientNamespace, string clientName, SourceInputModel sourceInputModel)
+        {
+            var clientSourceModel = sourceInputModel.FindForType(clientNamespace, clientName);
+            if (clientSourceModel != null)
+            {
+                var newClientMethods = new LowLevelClientMethod[clientMethods.Length];
+                for (int i = 0; i < clientMethods.Length; i++)
+                {
+                    newClientMethods[i] = ResolveClientMethod(clientMethods[i], clientSourceModel);
+                }
+                return newClientMethods;
+            }
+
+            return clientMethods;
+        }
+
+        /// <summary>
+        /// Reslove ambiguity between a DPG client method and client methods under the client with the same name as DPG client in existing codes.
+        /// </summary>
+        /// <param name="clientMethod">Client method belonging  to a DPG client.</param>
+        /// <param name="clientSourceModel">Roslyn model for a client with the same name of DPG client in existing codes.</param>
+        /// <returns>A LowLevelClientMethod which have been resolved to avoid ambiguity with the methods in the given client in existing codes.</returns>
+        private static LowLevelClientMethod ResolveClientMethod(LowLevelClientMethod clientMethod, INamedTypeSymbol clientSourceModel)
+        {
+            IEnumerable<IMethodSymbol> clientMethodSourceModels = GetPublicMethodsFromSourceModelByName(clientMethod.Signature.Name, clientSourceModel);
+            if (clientMethodSourceModels.Any())
+            {
+                return ResolveClientMethod(clientMethod, clientMethodSourceModels);
+            }
+            return clientMethod;
+        }
+
+        private static IEnumerable<IMethodSymbol> GetPublicMethodsFromSourceModelByName(string methodName, INamedTypeSymbol clientSourceModel)
+        {
+            return clientSourceModel.GetMembers().Where(s => s.Kind == SymbolKind.Method &&
+                s.Name == methodName && s.DeclaredAccessibility == Accessibility.Public).Select(s => (IMethodSymbol)s);
```

Comment:
`DeclaredAccessibility` can have any value here, not just `Public`

Reply:
@AlexanderSher Correct. But we're generating a public method (e.g. `ClientMethod`), and the ambiguity should happen from customers codes which invoke the public method. If you have an internal/protected/private ambiguous method, it won't cause compilation error in customers codes.

Like this:　https://sharplab.io/#v2:CYLg1APgAgTAjAWAFBQMwAJboMLoN7LpGYZRwBsmALOgLYCGAlgHYAUZADANoC669AJwDOASnyFikgG6D0AI3QBedMwCmAd3QAhViIDcEyUTkA6ALKs4+w+gC+ye0mRpMMbcgJJJLqDQssAF34AGnRAgH55JRUAVwAbOLFPSUdJGwB6dP5aOUYAcxiAexihMIAzdABjAAt6ZjzVYHQAwpIbAAcBQoDVSp6m33R/ZiD6UM4o5WZ4xPEvYkdbIA===

Reply:
I think removing `DeclaredAccessbility == Accessiblity.Public` do no harm here. I just want to avoid over-resolving the ambiguity.